### PR TITLE
Return sync cache

### PR DESCRIPTION
**Pull request by rem1niscence**
adds boolean cache field to return on sync/chain checks

### CODE REVIEW

File: src/services/sync-checker.ts [hunk 1]
```diff
@@ -48,23 +49,25 @@ export class SyncChecker {
     const syncedNodesKey = `sync-check-${sessionHash}`
     const syncedNodesCached = await this.redis.get(syncedNodesKey)
 
-    if (syncedNodesCached) {
+    const cached = !!syncedNodesCached
```
Comment (nymd) [inline src/services/sync-checker.ts]: What is this nonsense double negative type conversion

**Review by nymd — APPROVED** [review]

File: src/services/sync-checker.ts (hunk 1 above)
Comment (rem1niscence) [reply src/services/sync-checker.ts]: welcome to js-land where nothing  works as expected, this double negative is no error, is just a way of trully converting a value that is "falsey" to a fully boolean, on this case, `syncedNodesCached` could come as `null` or string which are different falsey and not falsey datatypes respectively but can throw unexpected errors if treated as a fully boolean, with double exclamation mark we are forcing the null value to become a true boolean value. For more info check [this](https://stackoverflow.com/questions/29312123/how-does-the-double-exclamation-work-in-javascript)

Comment (Evalir) [reply src/services/sync-checker.ts]: This is old stuff @rem1niscence . You can just do 
```suggestion
    const cached = Boolean(syncedNodesCached)
```

That answer is from 2015. Don't blame the language. :)

**Comment by rem1niscence** [comment]
ez

**Pull request merged**